### PR TITLE
Feature/bus 825 youtube video cms

### PR DESCRIPTION
**Pull request by carojansing**
Adds Youtube CMS Element

### CODE REVIEW

File: components/cms/element/CmsElementYoutubeVideo.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const youtubeVideo = ref(null);
+
+const videoId = props.element.config.videoID.value
+const videoUrl = computed(() => `https://www.youtube.com/embed/${videoId}`)
+</script>
+
+<template>
+    <div ref="youtubeVideo" class="my-0 mx-auto">
+        <div class="youtube-player h-full w-full relative ">
```
Comment (mloeffle) [inline components/cms/element/CmsElementYoutubeVideo.vue:16]: couldn't these two divs be combined?

File: components/cms/element/CmsElementProductBox.vue [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const props = defineProps<{
+    element: CmsElementProductBox;
+}>();
+
+const { product } = useProduct(props.element.data.product);
+</script>
+
+<template>
+    <ProductCard :product="product"></ProductCard>
```
Comment (mloeffle) [inline components/cms/element/CmsElementProductBox.vue:12]:
```suggestion
    <ProductCard :product="product" />
```

File: components/cms/block/CmsBlockYoutubeVideo.vue [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<script setup lang="ts">
+import type { Schemas } from '@shopware/api-client/api-types';
+const props = defineProps<{
+    block: Schemas['CmsBlock'];
+}>();
+const { getSlotContent } = useCmsBlock(props.block);
+const video: Schemas['CmsSlot'] = getSlotContent('video');
+</script>
```
Comment (aheartforspinach) [inline components/cms/block/CmsBlockYoutubeVideo.vue:8]:
```suggestion
<script setup lang="ts">
import type { Schemas } from '@shopware/api-client/api-types';

const props = defineProps<{
    block: Schemas['CmsBlock'];
}>();

const { getSlotContent } = useCmsBlock(props.block);
const video: Schemas['CmsSlot'] = getSlotContent('video');
</script>
```

File: components/cms/element/CmsElementYoutubeVideo.vue [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                :src="videoUrl"
+                frameborder="0"
+                class="w-full h-full absolute top-0 left-0"
+                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; fullscreen"
+                allowfullscreen
+            ></iframe>
+        </div>
+    </div>
+</template>
+<style scoped>
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementYoutubeVideo.vue:27]:
```suggestion
</template>

<style scoped>
```

File: components/cms/element/CmsElementYoutubeVideo.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+<template>
+    <div ref="youtubeVideo" class="my-0 mx-auto youtube-player h-full w-full relative">
+            <iframe
+                :src="videoUrl"
+                frameborder="0"
+                class="w-full h-full absolute top-0 left-0"
+                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; fullscreen"
+                allowfullscreen
+            ></iframe>
+        </div>
```
Comment (mloeffle) [inline components/cms/element/CmsElementYoutubeVideo.vue:23]:
```suggestion
    <div ref="youtubeVideo" class="my-0 mx-auto youtube-player h-full w-full relative">
        <iframe
            :src="videoUrl"
            frameborder="0"
            class="w-full h-full absolute top-0 left-0"
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; fullscreen"
            allowfullscreen
        />
    </div>
```

**Review by aheartforspinach — APPROVED** [review]

**Pull request merged**